### PR TITLE
Feat/board paste

### PR DESCRIPTION
**Pull request by DiamondThree**

### CODE REVIEW

File: libs/components/editor-core/src/editor/editor.ts [hunk 1]
```diff
@@ -479,7 +479,22 @@ export class Editor implements Virgo {
...
+    public copyBlock = async (blockId: string, parentId: string) => {
+        let block = await this.getBlockById(blockId);
+        let newBlcok: AsyncBlock;
+        newBlcok = await this.createBlock(block.type, parentId);
+        let parentProperties = await block.getProperties();
+        newBlcok.setProperties(parentProperties);
+        let children = await block.children();
+        if (children.length) {
+            await Promise.all(
+                children.map(async child => {
```
Comment (lawvs) [inline libs/components/editor-core/src/editor/editor.ts:491]: The `map` method has semantics that returns something

```suggestion
                children.forEach(async child => {
```

File: libs/components/editor-core/src/editor/editor.ts [hunk 2]
```diff
@@ -479,7 +479,22 @@ export class Editor implements Virgo {
...
+                    this.copyBlock(child.id, newBlcok.id);
+                })
+            );
```
Comment (lawvs) [inline libs/components/editor-core/src/editor/editor.ts:494]: It seems you forgot to await or return a promise at this block

File: libs/components/editor-core/src/editor/editor.ts [hunk 3]
```diff
@@ -479,7 +479,22 @@ export class Editor implements Virgo {
             return false;
         });
     }
-
+    public copyBlock = async (blockId: string, parentId: string) => {
```
Comment (lawvs) [inline libs/components/editor-core/src/editor/editor.ts:482]: Is the parent id is optional?

Comment (QiShaoXuan) [reply libs/components/editor-core/src/editor/editor.ts:482]: This doesn't look like a copy of the block, but rather a new blcock created and inserted into the parent block

**Pull request closed without merging**